tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/views/graph_executions/graph_executions_container_test.ts

```diff
@@ -167,4 +183,53 @@ describe('Graph Executions Container', () => {
     expect(tensorNames.length).toBe(0);
     expect(opTypes.length).toBe(0);
   }));
+
+  for (const oldFocusIndex of [null, 0, 119]) {
+    for (const newFocusIndex of [1, 60, 100, 118]) {
+      it(
+        `calls scrollToIndex on focusIndex change: ` +
+          `${oldFocusIndex} --> ${newFocusIndex}`,
+        fakeAsync(() => {
+          const fixture = TestBed.createComponent(GraphExecutionsComponent);
+          const component = fixture.componentInstance;
+          const graphExecutionData: {[index: number]: GraphExecution} = {};
+          const graphExecutionIndices: number[] = [];
+          for (let i = 0; i < 120; ++i) {
+            graphExecutionData[i] = createTestGraphExecution({
+              op_name: `TestOp_${i}`,
+              op_type: `OpType_${i}`,
+              tensor_debug_mode: TensorDebugMode.CONCISE_HEALTH,
+              debug_tensor_value: [i, 100, 0, 0, 0],
+            });
+            graphExecutionIndices.push(i);
+          }
+          component.numGraphExecutions = 120;
+          component.graphExecutionData = graphExecutionData;
+          component.graphExecutionIndices = graphExecutionIndices;
+          component.focusIndex = oldFocusIndex;
+          fixture.detectChanges();
+          tick();
+
+          const viewPort = component.TEST_ONLY.getViewPort() as CdkVirtualScrollViewport;
+          const {start, end} = viewPort.getRenderedRange();
+          expect(end).toBeGreaterThan(start);
+          const scrollIndices: number[] = [];
+          const scrollToIndexSpy = spyOn(
+            viewPort,
+            'scrollToIndex'
+          ).and.callFake((scrollIndex: number) => {
+            scrollIndices.push(scrollIndex);
+          });
+
+          component.ngOnChanges({
+            focusIndex: new SimpleChange(oldFocusIndex, newFocusIndex, true),
+          });
```

Comment:
Instead of calling method like this, prefer to create a wrapper component in the test and set new property values to the wrapper.

Reply:
Done by using the `GraphExecutionContainer`, which naturally wraps `GraphExecutionComponent`. Thanks.